File: Telemedicine-and-Doctor-Consultation-Platform-main/backend/routes/analytics.js
const router = require('express').Router();
const auth   = require('../middleware/auth');
const db     = require('../config/db');

// GET /api/analytics/overview
router.get('/overview', auth, async (req, res) => {
  try {
    const [[{patients}]] = await db.query('SELECT COUNT(*) as patients FROM patients');
    const [[{doctors}]]  = await db.query('SELECT COUNT(*) as doctors FROM doctors');
    const [[{consultations}]] = await db.query('SELECT COUNT(*) as consultations FROM consultations');
    const [[{orders}]]   = await db.query('SELECT COUNT(*) as orders FROM orders');
    const [[{prescriptions}]] = await db.query('SELECT COUNT(*) as prescriptions FROM prescriptions');
    const [[{totalRevenue}]] = await db.query('SELECT SUM(total_amount) as totalRevenue FROM orders WHERE status != "cancelled"');
    const [[{lowStockItems}]] = await db.query('SELECT COUNT(*) as lowStockItems FROM inventory WHERE qty < min_qty');

    res.json({
      patients, doctors, consultations, orders, prescriptions, 
      totalRevenue: totalRevenue || 0, 
      lowStockItems
    });
  } catch (err) {
    res.status(500).json({ error: 'Database error', message: err.message });
  }
});

// GET /api/analytics/consultations/weekly
router.get('/consultations/weekly', auth, (_req, res) => {
  // Simulated weekly data for demo
  res.json({
    labels: ['Mon','Tue','Wed','Thu','Fri','Sat','Sun'],
    data:   [42, 58, 50, 74, 66, 30, 18],
    total:  338,
    avgPerDay: 48,
  });
});

// GET /api/analytics/revenue/monthly
router.get('/revenue/monthly', auth, (_req, res) => {
  res.json({
    labels: ['Jan','Feb','Mar','Apr'],
    data:   [280000, 310000, 352000, 420000],
    growth: '+19%',
  });
});

// GET /api/analytics/doctors/performance
router.get('/doctors/performance', auth, async (req, res) => {
  try {
    const [rows] = await db.query('SELECT id, name, speciality, rating, total_patients FROM doctors');
    // For demo purposes, we extrapolate revenue
    res.json(rows.map(d => ({
      id: d.id, name: d.name, speciality: d.speciality,
      consultations: Math.floor(Math.random() * 50) + 10,
      avgDuration: (Math.random() * 10 + 10).toFixed(1),
      rating: d.rating,
      revenue: Math.floor(Math.random() * 50000) + 20000,
    })));
  } catch (err) {
    res.status(500).json({ error: 'Database error', message: err.message });
  }
});

// GET /api/analytics/pharmacy/revenue
router.get('/pharmacy/revenue', auth, (_req, res) => {
  res.json({
    today:    18400,
    thisWeek: 126000,
    mtd:      420000,
    breakdown: {
      consultationFees: 180000,
      pharmacy:         210000,
      labReports:       30000,
    },
  });
});

module.exports = router;

File: backend/routes/analytics.py
from flask import Blueprint, request, jsonify
from config import get_db_connection
from middleware.auth import token_required
import random

bp = Blueprint('analytics', __name__, url_prefix='/api/analytics')

@bp.route('/overview', methods=['GET'])
@token_required
def get_overview():
    try:
        conn = get_db_connection()
        cursor = conn.cursor()
        
        # Get counts
        cursor.execute('SELECT COUNT(*) as count FROM patients')
        patients = cursor.fetchone()['count']
        
        cursor.execute('SELECT COUNT(*) as count FROM doctors')
        doctors = cursor.fetchone()['count']
        
        cursor.execute('SELECT COUNT(*) as count FROM consultations')
        consultations = cursor.fetchone()['count']
        
        cursor.execute('SELECT COUNT(*) as count FROM orders')
        orders = cursor.fetchone()['count']
        
        cursor.execute('SELECT COUNT(*) as count FROM prescriptions')
        prescriptions = cursor.fetchone()['count']
        
        cursor.execute('SELECT SUM(total_amount) as total FROM orders WHERE status != "cancelled"')
        total_revenue_row = cursor.fetchone()
        total_revenue = float(total_revenue_row['total'] or 0)
        
        cursor.execute('SELECT COUNT(*) as count FROM inventory WHERE qty < min_qty')
        low_stock_items = cursor.fetchone()['count']
        
        cursor.close()
        conn.close()
        
        return jsonify({
            'patients': patients,
            'doctors': doctors,
            'consultations': consultations,
            'orders': orders,
            'prescriptions': prescriptions,
            'totalRevenue': total_revenue,
            'lowStockItems': low_stock_items
        }), 200
    
    except Exception as err:
        return jsonify({'error': 'Database error', 'message': str(err)}), 500

@bp.route('/consultations/weekly', methods=['GET'])
@token_required
def get_weekly_consultations():
    return jsonify({
        'labels': ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'],
        'data': [42, 58, 50, 74, 66, 30, 18],
        'total': 338,
        'avgPerDay': 48,
    }), 200

@bp.route('/revenue/monthly', methods=['GET'])
@token_required
def get_monthly_revenue():
    return jsonify({
        'labels': ['Jan', 'Feb', 'Mar', 'Apr'],
        'data': [280000, 310000, 352000, 420000],
        'growth': '+19%',
    }), 200

@bp.route('/doctors/performance', methods=['GET'])
@token_required
def get_doctor_performance():
    try:
        conn = get_db_connection()
        cursor = conn.cursor()
        cursor.execute('SELECT id, name, speciality, rating, total_patients FROM doctors')
        doctors = cursor.fetchall()
        cursor.close()
        conn.close()
        
        result = []
        for d in doctors:
            result.append({
                'id': d['id'],
                'name': d['name'],
                'speciality': d['speciality'],
                'consultations': random.randint(10, 60),
                'avgDuration': f'{random.uniform(10, 20):.1f}',
                'rating': d.get('rating', 4.5),
                'revenue': random.randint(20000, 70000),
            })
        
        return jsonify(result), 200
    
    except Exception as err:
        return jsonify({'error': 'Database error', 'message': str(err)}), 500

@bp.route('/pharmacy/revenue', methods=['GET'])
@token_required
def get_pharmacy_revenue():
    return jsonify({
        'today': 18400,
        'thisWeek': 126000,
        'mtd': 420000,
        'breakdown': {
            'consultationFees': 180000,
            'pharmacy': 210000,
            'labReports': 30000,
        },
    }), 200
